site/content/blog/kapp-versioned-resource.md

```diff
@@ -0,0 +1,566 @@
+  ...
+  1,  1   data:
+  2     -   hello_msg: hello-kapp
+      2 +   hello_msg: hello-carvel
+  3,  3   kind: ConfigMap
+  4,  4   metadata:
+
+Changes
+
+Namespace  Name              Kind       Age  Op      Op st.  Wait to    Rs  Ri  
+default    crd-config-ver-2  ConfigMap  -    create  -       reconcile  -   -  
+
+Op:      1 create, 0 delete, 0 update, 0 noop, 0 exists
+Wait to: 1 reconcile, 0 delete, 0 noop
+
+Continue? [yN]:y
+```
+
+
+Now what we want is whenever there is any change happen in configmap `crd-config` the custom resource `first-cr` should get updated or restarted. To achive this we have to add annotation `kapp.k14s.io/versioned-explicit-ref` to the custom resource so that on every update in configmap `crd-config` kapp will make update to custom resource as well.
+
+After adding `kapp.k14s.io/versioned-explicit-ref` to custom resource `first-cr`, it will be something like:
+```yaml
+apiVersion: "example.com/v1beta1"
+kind: simplecrd
+metadata:
+  name: first-cr
+  annotations:
+    kapp.k14s.io/versioned-explicit-ref: |
+      apiVersion: v1
+      kind: ConfigMap
+      name: crd-config
+spec:
+  name: simple-app
+  image: docker.io/dkalinin/k8s-simple-app:latest
+```
+
+Let's deploy updated YAML `version-crd.yaml`.
+```bash
+$ kapp deploy -a crd-app -f version-crd.yaml               
+Target cluster 'https://127.0.0.1:33907' (nodes: minikube)
+
+Changes
+
+Namespace  Name      Kind       Age  Op      Op st.  Wait to    Rs  Ri  
+default    first-cr  simplecrd  3d   update  -       reconcile  ok  -  
+
+Op:      0 create, 0 delete, 1 update, 0 noop, 0 exists
+Wait to: 1 reconcile, 0 delete, 0 noop
+
+Continue? [yN]: y
+```
+
+Let's update the value of `data.hello_msg` to `hello-tanzu` in configmap `crd-config` and re-deploy the app `crd-app`.
+
+```bash
+$ kapp deploy -a app1 -f version-crd.yaml --diff-changes
+Target cluster 'https://127.0.0.1:33907' (nodes: minikube)
+
+@@ create configmap/crd-config-ver-3 (v1) namespace: default @@
+  ...
+  1,  1   data:
+  2     -   hello_msg: hello-carvel
+      2 +   hello_msg: hello-tanzu
+  3,  3   kind: ConfigMap
+  4,  4   metadata:
+@@ update simplecrd/first-cr (example.com/v1beta1) namespace: default @@
+  ...
+  6,  6         kind: ConfigMap
+  7     -       name: crd-config-ver-2
+      7 +       name: crd-config-ver-3
+  8,  8     creationTimestamp: "2022-06-17T22:34:53Z"
+  9,  9     generation: 1
+
+Changes
+
+Namespace  Name              Kind       Age  Op      Op st.  Wait to    Rs  Ri  
+default    crd-config-ver-3  ConfigMap  -    create  -       reconcile  -   -  
+^          first-cr          simplecrd  3d   update  -       reconcile  ok  -  
+
+Op:      1 create, 0 delete, 1 update, 0 noop, 0 exists
+Wait to: 2 reconcile, 0 delete, 0 noop
+
+Continue? [yN]: y
+```
+
+After adding annotion `kapp.k14s.io/versioned-explicit-ref` to the custom resource `first-cr`, kapp is able to make update to it whenever there is any new changes in configmap `crd-config`.
+
+
```

Comment:
There seems to be a sudden ending to this blog post and not a wrap up / in conclusion piece. Can you write a few sentences to conclude the post before the "Join us on Slack"??